api/src/main/java/jakarta/data/Insert.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>The {@code Insert} annotation indicates that the annotated repository method requests that one or more entities
+ * be inserted into the database. This method must have a single parameter whose type must be one of the following:
+ * </p>
+ * <ul>
+ *     <li>The entity to be inserted.</li>
+ *     <li>An {@code Iterable} of entities to be inserted.</li>
+ *     <li>An array of entities to be inserted.</li>
+ * </ul>
+ * <p>The return type of the annotated method should match the type of the parameter. For example, if the method is
+ * annotated with {@code @Insert} and takes a parameter of type {@code Car car}, the return type should be {@code Car}.
+ * Similarly, if the parameter is an {@code Iterable<Car>} or an array of {@code Car}, the return type should be the
+ * corresponding type.
+ * </p>
+ * <p>After invoking this method, it is recommended not to use the entity value supplied as a parameter, as this method
+ * makes no guarantees about the state of the entity value after insertion.
+ * </p>
+ * <p>If the entity uses optimistic locking, and the version differs from the version in the database, an
+ * {@link jakarta.data.exceptions.OptimisticLockingFailureException} may be thrown.
+ * </p>
+ * <p>For example, consider an interface representing a garage:</p>
+ * <pre>
+ * {@literal @}Repository
+ * interface Garage {
+ *     {@literal @}Insert
+ *     Car parking(Car car);
+ * }
+ * </pre>
+ * <p>The {@code @Insert} annotation can be used to indicate that the {@code parkCar} method is responsible for inserting
+ * a car entity into a database.
+ * </p>
+ *
+ * <p>If this annotation is combined with other operation annotations (e.g., {@code @Update}, {@code @Delete},
+ *  {@code @Save}), it will throw an {@link IllegalStateException} as only one operation type can be specified, where
+ *  a Jakarta Data provider implementation choose to detect (and report) this error at compile time or at runtime..</p>
```

Comment:
Thank you for pointing it out, Nathan.
But at the it shows: **[UnsupportedOperationException](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/UnsupportedOperationException.html)**

> Thrown to indicate that the requested operation is not supported.

What I understand: Both annotations are supported, but in this case the user is doing something wrong! Should we have an exception for this?
What do you think?

Reply:
> Thank you for pointing it out, Nathan. But at the it shows: **[UnsupportedOperationException](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/UnsupportedOperationException.html)**
> 
> > Thrown to indicate that the requested operation is not supported.
> 
> What I understand: Both annotations are supported, but in this case the user is doing something wrong! Should we have an exception for this? What do you think?

The Jakarta Data provider would need to include a meaningful message for it to make sense, for example:

`UnsupportedOperationException: The YourRepository.doSomething repository method has conflicting annotations: @Save and @Delete.`

It would be fine with me if we instead introduce a new exception type for this. For example,

`AnnotationConflictException: @Save and @Delete cannot be combined on the YourRepository.doSomething repository method.`

My point here was just that `IllegalStateException` doesn't fit.

Reply:
I checked in the exception present in Java; the IllegalArgumentException was the closest that makes sense for us.

Maybe create one might sound better, such as IllegalConfigurationException...

Reply:
Ahhh, that makes sense.
thank you.